### PR TITLE
Native beacon state: v3

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

This PR modifies code inside `state/state-native/v3` in a similar fashion as #10121 did for `v2`.

**Which issues(s) does this PR fix?**

Extracted from #9820

### CODE REVIEW

File: beacon-chain/state/state-native/v3/state_trie.go
Comment (nisdas) [inline beacon-chain/state/state-native/v3/state_trie.go:114]: Change this back

File: beacon-chain/state/state-native/v3/state_trie.go
Comment (nisdas) [inline beacon-chain/state/state-native/v3/state_trie.go:236]: change back

File: beacon-chain/state/state-native/v3/beacon_state_mainnet.go
Comment (rauljordan) [inline beacon-chain/state/state-native/v3/beacon_state_mainnet.go:21]: We tend to follow the naming conventions `Bytes32` in the rest of the repository, so maybe we can align on consistency in a future PR

**Review by rauljordan — APPROVED** [review]

**Comment by rauljordan** [comment]
Needs a codeowner approval from @nisdas

**Review by nisdas — APPROVED** [review]

File: beacon-chain/state/state-native/v3/beacon_state_mainnet.go
Comment (rkapka) [reply beacon-chain/state/state-native/v3/beacon_state_mainnet.go:21]: This will be removed. See https://github.com/prysmaticlabs/prysm/issues/10100

**Pull request merged**
